fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -382,7 +385,7 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
             }
 
             // Validate the input and, if valid, start collecting supplemental warnings
-            List<Issue> warnings = new ArrayList<>(validateInput(newResource));
+            List<Issue> warnings = skipValidation ? new ArrayList<>() : new ArrayList<>(validateInput(newResource));
```

Comment:
should it just be the validateInput(newResource)'s output not passing to a new arraylist?

Reply:
I wanted to keep the semantics the same as the old code. I thought that warnings might be added to later on, hence the need to put it into its own list (the validateInput could be immutable). I think a better pattern would be to create the list, then pass it in for example: validateInput(warnings, newResource); The validateInput method could simply return the number of warnings it added.